cosmos/providers/databricks/plugin.py

```diff
@@ -0,0 +1,279 @@
+    :param task_map: A dictionary mapping task IDs to BaseOperator instances.
+    :return: A list of Databricks task IDs for the given task group.
+    """
+    task_ids = []
+    print("Getting task ids for db")
+    for task_id, task in task_map.items():
+        if task_id != f"{group_id}.launch":
+            print(f"databricks task id {_get_databricks_task_id(task)}")
+            task_ids.append(_get_databricks_task_id(task))
+    return task_ids
+
+
+def _repair_task(
+    databricks_conn_id: str, databricks_run_id: str, tasks_to_repair: list[str]
+) -> dict:
+    """
+    This function allows the Airflow retry function to create a repair job for Databricks.
+    It uses the Databricks API to get the latest repair ID before sending the repair query.
+
+    Note that we use the `JobsService` class instead of the `RunsApi` class. This is because the
+    `RunsApi` class does not allow sending the `include_history` parameter which is necessary for
+    repair jobs.
+
+    Also for the moment we don't allow custom retry_callbacks. We might implement this in
+    the future if users ask for it, but for the moment we want to keep things simple while the API
+    stabilizes.
+
+    :param databricks_conn_id: The Databricks connection ID.
+    :param databricks_run_id: The Databricks run ID.
+    :param tasks_to_repair: A list of Databricks task IDs to repair.
+    :return: None
+    """
+
+    def _get_api_client():
+        hook = DatabricksHook(databricks_conn_id)
+        databricks_conn = hook.get_conn()
+        return ApiClient(
+            user=databricks_conn.login,
+            token=databricks_conn.password,
+            host=databricks_conn.host,
+        )
+
+    api_client = _get_api_client()
+    jobs_service = JobsService(api_client)
+    current_job = jobs_service.get_run(run_id=databricks_run_id, include_history=True)
+    repair_history = current_job.get("repair_history")
+    repair_history_id = None
+    if repair_history and len(repair_history) > 1:
+        # We use the last item in the array to get the latest repair ID
+        repair_history_id = repair_history[-1]["id"]
+    return jobs_service.repair(
+        run_id=databricks_run_id,
+        version="2.1",
+        latest_repair_id=repair_history_id,
+        rerun_tasks=tasks_to_repair,
+    )
+
+
+class DatabricksJobRunLink(BaseOperatorLink):
+    """Constructs a link to monitor a Databricks Job Run."""
+
+    name = "See Databricks Job Run"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        dag = get_airflow_app().dag_bag.get_dag(ti_key.dag_id)
+        dag.get_task(ti_key.task_id)
+        # Should we catch the exception here if there is no return value?
+        metadata = XCom.get_one(
+            task_id=ti_key.task_id,
+            dag_id=ti_key.dag_id,
+            run_id=ti_key.run_id,
+            key="return_value",
+        )
```

Comment:
Can we not use the below? It looks more convenient method as per Airflow's docstring for `get_one` implementation.
```suggestion
        metadata = XCom.get_value(
            ti_key=ti_key,
            key="return_value",
        )
```

Ref: [databricks/support-for-retries ](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/xcom/index.html#airflow.models.xcom.BaseXCom.get_value)

Reply:
Oh neat didn't know this was a thing. Thanks @pankajkoti !